nagios.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// NagiosData is a task factory for tasks that fetch a JSON representation of
+// the provided project. The task uses outFor and errOutFor to get io.Writers
+// to write, respectively, the JSON output and any eventual error message.
+func NagiosData(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return nagiosData(func(project, pod string) *exec.Cmd {
+		return exec.Command("oc", "exec", pod, "cat", "/var/log/nagios/status.dat")
+	}, project, outFor, errOutFor)
+}
+
+// A getNagiosDataCmdFactory generates commands to get the raw nagios status data
+// in the provided project from the provided pod
+type getNagiosDataCmdFactory func(project, pod string) *exec.Cmd
+
+// nagiosData will dump a JSON representation of the current nagios status into
+// the io.Writer returned from outFor and log any errors it encounters doing so
+// into the io.Writer returned from errOutFor.
```

Comment:
We could accidentally introduce errors here. I'd opt for keeping the "run a command and save the output", that will give us the Nagios logs in its original format.

Yes we might need to parse that later as part of the analysis, but I'd suggest we do it then, not in the dump part.
Save the code, it will be useful soon :)


BTW, is this .dat file an ever growing file with everything since the first run?